go/private/tools/path.bzl

```diff
@@ -77,7 +77,8 @@ def _go_path_impl(ctx):
         src_dir = None
 
         for f in pkg.srcs:
-            src_dir = f.dirname
+            if src_dir == None:
+                src_dir = f.dirname
```

Comment:
I am not very familiar with `go_path`. Could you explain why using the first rather than the last source directory is the correct choice?

Reply:
Sure, as you can see here https://github.com/bazelbuild/rules_go/blob/master/go/private/context.bzl#L237, the `attr.srcs` files are added before the `generated_srcs` which means the relative paths are always first, the unsolved case is when all the files are coming from the `generated_srcs` and `attr_srcs` is empty because none of the files is relative

Reply:
That makes sense, thanks for the pointer.

Could it be that the root of the problem is that we are separately storing all the potential roots (dirnames of source files) and potentially embedded files (embedsrcs) but not the association between the two (which dirnames of sources were present on the target that defined a particular embedsrc). It seems to me that the correct approach could be to store this association in the provider and then make all `embedsrcs` of a particular target relative to all dirnames of source files of that target.

@bakjos What do you think?

Reply:
@fmeum it does makes sense, let me take another pass, because, before https://github.com/bazelbuild/rules_go/pull/3285 was working well if not will try to include the map in the provider, thanks for the guidance

Reply:
@fmeum Took some days off but added a new test case for generated sources only and a root dir per GoSource and seems to work